framework/src/application.ts

```diff
@@ -233,52 +232,39 @@ export class Application {
 		this._blockchainDB = this._getDBInstance(this.config, 'blockchain.db');
 		this._nodeDB = this._getDBInstance(this.config, 'node.db');
 
-		// Acquire mutex for loading process
-		this._runMutexReleaseHandler = await this._mutex.acquire();
+		await this._mutex.runExclusiveWithTimeout<void>(
+			async () => {
+				// Initialize all objects
+				this._channel = this._initChannel();
 
-		// Wait for RUN_WAIT_TIMEOUT to finish
-		setTimeout(() => {
-			// If mutex is still locked
-			if (this._mutex.isLocked()) {
-				// Release mutex
-				this._runMutexReleaseHandler();
+				this._controller = this._initController();
 
-				// Throw exception so it can shutdown application
-				throw new Error(`Application could not started in ${RUN_TIMEOUT}ms`);
-			}
-		}, RUN_TIMEOUT);
-
-		// Initialize all objects
-		this._channel = this._initChannel();
-
-		this._controller = this._initController();
+				await this._controller.load();
 
-		await this._controller.load();
+				await this._node.init({
+					bus: this._controller.bus,
+					channel: this._channel,
+					forgerDB: this._forgerDB,
+					blockchainDB: this._blockchainDB,
+					nodeDB: this._nodeDB,
+					logger: this.logger,
+				});
 
-		await this._node.init({
-			bus: this._controller.bus,
-			channel: this._channel,
-			forgerDB: this._forgerDB,
-			blockchainDB: this._blockchainDB,
-			nodeDB: this._nodeDB,
-			logger: this.logger,
-		});
+				await this._controller.loadPlugins(this._plugins, this.config.plugins);
+				this.logger.debug(this._controller.bus.getEvents(), 'Application listening to events');
+				this.logger.debug(this._controller.bus.getActions(), 'Application ready for actions');
 
-		// Release mutex
-		this._runMutexReleaseHandler();
-
-		await this._controller.loadPlugins(this._plugins, this.config.plugins);
-		this.logger.debug(this._controller.bus.getEvents(), 'Application listening to events');
-		this.logger.debug(this._controller.bus.getActions(), 'Application ready for actions');
-
-		this._channel.publish('app:ready');
+				this._channel.publish('app:ready');
+			},
+			RUN_TIMEOUT,
```

Comment:
as we don't know the initalization time required for  mainnet, or other network, i think it's safer to not to include the timeout here.
Also, even if it fails to start, it shouldnt be a problem if killed